### ISSUE TITLE
Unexpected window when stream is shorter than desired window size

### ISSUE THREAD

**Issue by magzot** [poetix/protonpack#44]
The documentation of windowed says that the last window is guaranteed to be of the desired size (possible data loss). Passing in a stream with less than window size elements however results in a window smaller than the desired size. 

In the example below it would be nice if one doesn't get a window of size 1, but rather no window at all.

```
IntStream stream = IntStream.of(1);
List<Integer> sumOfPairs = StreamUtils.windowed(stream.boxed(), 2)
                .map(x -> x.get(0) + x.get(1))
                .collect(Collectors.toList());
```

**Comment by thall** [comment]
Nice, i would really like to get this patch in, it solves my problems with `windowed`. @poetix

**Comment by poetix** [comment]
Merged and released - v1.16 now in Maven Central.

**Comment by magzot** [comment]
Nice!

ons 22 jan. 2020 kl. 21:13 skrev Dominic Fox <notifications@github.com>:
